src/internal/client.ts

```diff
@@ -2400,4 +2402,31 @@ export class TypedClient {
 
     await this.makeRequestAsyncOmit({ method, bucketName, query }, '', [204])
   }
+
+  async getObjectRetention(
+    bucketName: string,
+    objectName: string,
+    getOpts?: GetObjectRetentionOpts | undefined,
```

Comment:
Do we need the `undefined` type here as we already have a question mark after `getOpts`? If not, I think we can apply the following:

```suggestion
    getOpts?: GetObjectRetentionOpts,
```